File: sol/2554/2554.py
class Solution:
    def maxCount(self, banned: List[int], n: int, maxSum: int) -> int:
        res = 0
        currSum = 0
        banned = frozenset(banned)
        for i in range(1, n + 1):
            if i in banned:
                continue
            if currSum + i > maxSum:
                break
            currSum += i
            res += 1
        return res

File: sol/2554/2554.cpp
struct Solution {
    int maxCount(vector<int>& banned, int n, int maxSum) {
        int res = 0, currSum = 0;
        vector<bool> banSet(1e4 + 1, false);
        for (int x : banned){
            banSet[x] = true;
        }
        for (int i = 1; i <= n; ++i){
            if (banSet[i]){
                continue;
            }
            if (currSum + i > maxSum){
                break;
            }
            currSum += i;
            ++res;
        }
        return res;
    }
};
